src/client/core/prevent-real-events.js

```diff
@@ -0,0 +1,84 @@
+    'pointermove', 'pointerover', 'pointerout', 'pointerup'
+];
+
+const KEY_EVENTS               = ['keydown', 'keypress', 'keyup'];
+const DEVTOOLS_HOTKEY_KEYCODES = [67, 73, 74, 75];                // 'C', 'I', 'J', 'K'
+const F12_KEY_CODE             = 123;
+
+// NOTE: when test is run we should block real events (from mouse, keyboard), because it may lead to
+// unexpected test result.
+function preventRealEventHandler (e, dispatched, preventDefault) {
+    var target = e.target || e.srcElement;
+
+    if (!dispatched && !isShadowUIElement(target)) {
+        //If an element loses focus because of becoming invisible, a blur event is raised. We must not prevent this blur event.
+        //In the IE an element loses focus only if css display property is set to 'none', other ways of making
+        // element invisible don't lead to blurring (in MSEdge focus/blur is sync)
+        if (e.type === 'blur') {
+            if (browserUtils.isIE && browserUtils.version < 12) {
+                var isElementInvisible = !isWindow(target) && get(target, 'display') === 'none';
+                var elementParents     = null;
+                var invisibleParents   = false;
+
+                if (!isElementInvisible) {
+                    elementParents   = getParents(target);
+                    invisibleParents = filter(elementParents, parent => get(parent, 'display') === 'none');
+                }
+
+                if (isElementInvisible || invisibleParents.length) {
+                    //B254768 - reason of setTimeout method using
+                    window.setTimeout(() => {
+                        eventSimulator.blur(target);
+                    }, 0);
+                }
+            }
+            //NOTE: fix for jQuery bug. An exception raises when call .is(':visible') for window or document on page loading (e.ownerDocument is null)
+            else if (target !== window && target !== window.document && !hasDimensions(target))
+                return;
+        }
+
+        preventDefault();
+    }
+}
+
+
+function hotkeyHandler (e, dispatched, preventEvent, cancelHandlers, stopEventPropagation) {
+    if (dispatched)
+        return;
+
+    // NOTE: To enable hotkeys we should allow specified keyboard events to pass the capture phase.
+    // Most browsers listen for keydown and keyup events, meanwhile Firefox and Safari will not process
+    // the hotkey combination until the keypress event will be raised (this event will be raised, if we
+    // called stopEventPropagation() for our specific keydown and keyup events before).
+
+    var buttonF12pressed = e.keyCode === F12_KEY_CODE;
+    var hotkeyShortcut   = (e.type === 'keydown' || e.type === 'keyup') &&
+                           DEVTOOLS_HOTKEY_KEYCODES.indexOf(e.keyCode) > -1 &&
+                           ((e.shiftKey && e.ctrlKey) || e.altKey);
+    var isKeypressEvent  = e.type === 'keypress' && (browserUtils.isFirefox || browserUtils.isSafari);

```

Comment:
Hm, I just saw it. In fact you don't prevent all of `keypress` event in firefox and safari? Even if it just `enter` key or any other?